### DISCUSSION TITLE
Custom theme not being applied

### DISCUSSION THREAD

**Discussion by HadiSDev** [saadeghi/daisyui#1644] [Q&A] [answered]
Hi.
I am new to Daisy UI and I am having trouble setting a custom theme. The default dark theme get applied no matter what:

https://stackblitz.com/edit/daisyui-solidjs-gjfm3f?file=src%2FApp.jsx

I just followed the docs for setting a new custom theme, but I don't know what I am missing

**Comment by HadiSDev** [comment, marked as answer]
Sorry. I forgot but "themes" under daisyui: {...} config. My bad

**Reply by kevinXmichael** [reply]
This is actually not working. Changing a color in your stackblitz project, for example primary, will not update it accordingly in the UI.

**Reply by saadeghi** [reply]
@kevinXmichael Did you save the file and did you run `npm run dev`?  
It works as expected if the config is correct:

Here's a fork and you can see @HadiSDev's custom theme:  
https://stackblitz.com/edit/daisyui-solidjs-gatzm8?file=tailwind.config.js

**Reply by kevinXmichael** [reply]
You are right! You fork works fine, my issue was the following: I had a Button component which calculated the color dynamically:
`className={"btn btn-${props.color ?? 'primary'}"}`
The issue is that the tailwind compiler will not recognise this so I changed it to always provide full class names:
`className={"btn ${props.color ?? 'btn-primary'}"}`
With this approach every time a pass a color the compiler will now to provide these classes.
